draft-aegis-mimi-arch-latest.md

```diff
@@ -348,11 +353,13 @@ ApplicationID is defined as a UUIDv4 string and MUST be unique across all
 gateways that are federated with one another. // TODO: Link to UUIDv4 definition
 
 ~~~
-struct {
-    UUIDv4 id;
-    IdentityConfiguration identity_configuration;
+ApplicationConfiguration
+{
+    "identity_configuration": IdentityConfiguration,
+    "id": UUIDv4
+    // TODO: Could apps use multiple IdentityConfigurations?
```

Comment:
yeah... not sure yet. clearly somethings still missing.

But i do think it might be interesting. Like when federating we may want to use different IDPoviders.

Maybe clients should include the identity_provider_type of their provider when they first register with the gateway?

Reply:
Could the same be achieved using compound ID providers? If we have providers P1 and P2, the app can use provider P who forwards stuff to P1 or P2 based on domain (or something)